sp_dev.py

```diff
@@ -124,5 +130,11 @@ def __init__(self, birth_topic, metrics):
 
     def add_dev(self, device):
         """Adds a device to EdgeNode."""
-        # TODO: check if device does not already exist.
-        self.devices.append(device)
+        if device in self.devices:
+            print("Device %s already exists" % dev.get_short_handle())
+        else:
+            self.devices.append(device)
+
+    def remove_dev(self, device):
+        """Removes a device from EdgeNode."""
+        self.devices = [dev for dev in self.devices if dev != device]
```

Comment:
`list` method have a remove method which may be more suitable and maybe a little bit more efficicent (but we're talking python here, so performance is not its strongest suit...)
```suggestion
        self.devices.remove(device)
```

Reply:
I changed it as suggested.